src/tiled/automappingutils.cpp

```diff
@@ -73,7 +77,8 @@ const QList<MapObject*> objectsInRegion(ObjectGroup *layer,
         // QRegion::intersects() returns false for empty regions even if they are
         // contained within the region, so we also check for containment of the
         // top left to include the case of zero size objects.
-        if (where.intersects(rect) || where.contains(rect.topLeft()))
+        // (Also, for Tile objects we need to check the bottom left)
+        if (where.intersects(rect) || where.contains(rect.topLeft()) || where.contains(rect.bottomLeft()))
```

Comment:
The problem here was that tile objects use the bottom left corner as their origin but I still think there are issues with this approach. Tile objects actually get bounding rectangles with a negative width and height of 1 which I think is complicating things. My next pull request will address this.